analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.cs

```diff
@@ -44,6 +46,52 @@ protected override ProgramState LearnBranchingConstraint(ProgramState state, IBi
         return state;
     }
 
+    protected override ProgramState PreProcess(ProgramState state, IBinaryOperationWrapper operation)
+    {
+        if (operation.LeftOperand is { } leftOperand
+            && operation.RightOperand is { } rightOperand
+            && state[leftOperand]?.Constraint<NumberConstraint>() is { } leftNumber
+            && state[rightOperand]?.Constraint<NumberConstraint>() is { } rightNumber
+            && Calculate(operation.OperatorKind, leftNumber, rightNumber) is { } newConstraint)
+        {
+            state = state.SetOperationConstraint(operation.WrappedOperation, newConstraint);
```

Comment:
Please add overload that will accept `IOperationWrapper`. There's already one for `SetOperationValue`.

We're still building what's handy to have.
```suggestion
            state = state.SetOperationConstraint(operation, newConstraint);
```